src/pages/index.js

```diff
@@ -33,10 +46,47 @@ constants.profileEditButton.addEventListener("click", () => {
   profileFormValidator.resetValidation();
 });
 
-function handleProfileFormSubmit({ name, description }) {
-  user.setUserInfo({ name, description });
+function handleProfileFormSubmit({ name, about }) {
+  profileEditModal.setLoading(true);
+  api
+    .updateUserProfile({ name, about })
+    .then(() => {
+      user.setUserInfo({ name, about });
+    })
+    .finally(() => {
+      profileEditModal.setLoading(false);
+      profileEditModal.close();
+    });
 }
 
+//User avatar
+const avatarEditModal = new ModalWithForm(
+  "#avatar-change-modal",
+  handleAvatarSubmit
+);
+avatarEditModal.setEventListeners();
+constants.userAvatarButton.addEventListener("click", () => {
+  avatarEditModal.open();
+});
+
+function handleAvatarSubmit({ link }) {
+  avatarEditModal.setLoading(true);
+  api
+    .updateAvatar({ link })
+    .then(() => {
+      user.setAvatar(link);
+    })
+    .finally(() => {
+      avatarEditModal.setLoading(false);
+      avatarEditModal.close();
+    });
+}
+
+api.getUserInfo().then((data) => {
```

Comment:
Better to use `Promise.all` to load initial data

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all